File: src/main/java/it/contrader/model/Library.java
package it.contrader.model;


import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name ="library")
public class Library {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String name;
	
	@OneToMany (mappedBy = "library")
	private List<Folder> folders;
	
	@OneToOne
	private Description description;
	
	@ManyToOne
	@JoinColumn(name ="id_interest")
	private Interest interest;
	
}

File: Project1/Model/Library.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Project1.Model
{
    public class Library : BaseModel
    {

        public string Name { get; set; }

        public int InterestId { get; set; }

        public Interest Interest { get; set; }

        public IList<Folder> Folders { get; set; }
    }
}